build-system/tasks/dist.js

```diff
@@ -55,6 +60,37 @@ const {maybeUpdatePackages} = require('./update-packages');
 const {green, cyan} = colors;
 const argv = require('minimist')(process.argv.slice(2));
 
+const babel = require('@babel/core');
+const deglob = require('globs-to-files');
+
+function transferSrcsToTempDir() {
+  const tmp = `${tempy.directory()}/amphtml`;
+  process.env.AMP_TMP_DIR = tmp;
+  log(green(`creating temp directory: ${tmp}`));
+
+  if (!isTravisBuild()) {
+    log('Writing transforms to', colors.cyan(tmp));
+  }
+  const files = deglob.sync(BABEL_SRC_GLOBS);
+  log(green('Executing babel transforms'));
+  files.forEach(file => {
+    if (file.startsWith('node_modules/') || file.startsWith('third_party/')) {
+      fs.copySync(file, `${tmp}/${file}`);
+      return;
+    }
+
+    const {code} = babel.transformFileSync(file, {
+      plugins: conf.plugins({
+        isEsmBuild: argv.esm,
+        isForTesting: argv.fortesting,
+      }),
+      retainLines: true,
+    });
```

Comment:
Optional: If you use `browserify` + `babelify` here, you can print progress dots for the babel transforms like this:

https://github.com/ampproject/amphtml/blob/b7610436de23dfd52a3d0a27ddc6a046c571a52c/build-system/tasks/runtime-test/runtime-test-base.js#L193-L199

Reply:
using browserify here would just be additional overhead and would add additional transformation that we might not need and that might conflict when we pass this to closure compiler.